src/sentry/sentry_apps/services/app/impl.py

```diff
@@ -76,6 +76,17 @@ def get_installation_by_id(self, *, id: int) -> RpcSentryAppInstallation | None:
         except SentryAppInstallation.DoesNotExist:
             return None
 
+    def get_installation_org_id_by_token_id(self, token_id: int) -> int | None:
+        filters: SentryAppInstallationFilterArgs = {
+            "status": SentryAppInstallationStatus.INSTALLED,
+            "api_installation_token_id": str(token_id),
+        }
+        queryset = self._FQ.apply_filters(SentryAppInstallation.objects.all(), filters)
+        install = queryset.first()
+        if not install:
+            return None
+        return install.organization_id
```

Comment:
This logic mirrors the current logic used in ratelimiting.

https://github.com/getsentry/sentry/blob/77f283353930f0b7abe05fbbfce1f78127c9a7d6/src/sentry/ratelimits/utils.py#L137-L154

I'll replace this logic with the new RPC call in a separate pull request because of deploy boundary issues.